test/system/src/test/java/io/pravega/test/system/StreamCutsTest.java

```diff
@@ -0,0 +1,279 @@
+    }
+
+    @After
+    public void tearDown() {
+        streamManager.close();
+        ExecutorServiceHelpers.shutdown(executor);
+    }
+
+    // Start utils region
+
+    @SneakyThrows
+    private <T extends Serializable> List<Map<Stream, StreamCut>> getStreamCutSlices(ClientFactory client, ReaderGroup readerGroup) {
+        @Cleanup
+        EventStreamReader<T> reader = client.createReader("slicer", readerGroup.getGroupName(), new JavaSerializer<>(),
+                ReaderConfig.builder().build());
+        List<Map<Stream, StreamCut>> streamCuts = new ArrayList<>();
+        EventRead<T> event;
+        int validEvents = 0;
+        try {
+            do {
+                event = reader.readNextEvent(5000);
+                if (event.getEvent() != null) {
+                    validEvents++;
+                } else {
+                    log.warn("Read unexpected null event at {}.", validEvents);
+                    continue;
+                }
+
+                // Get a StreamCut each defined number of events.
+                if (validEvents % CUT_SIZE == 0 && validEvents > 0) {
+                    reader.close();
+                    log.debug("Adding a StreamCut positioned at event {}.", validEvents);
+                    streamCuts.add(readerGroup.getStreamCuts());
+                    reader = client.createReader("slicer", readerGroup.getGroupName(), new JavaSerializer<>(),
+                            ReaderConfig.builder().build());
+                }
+            } while (validEvents < TOTAL_EVENTS);
+        } finally {
+            reader.close();
+        }
+
+        return streamCuts;
+    }
+
+    private void writeDummyEvents(ClientFactory clientFactory, String streamName, int totalEvents) {
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName, new JavaSerializer<>(),
+                EventWriterConfig.builder().build());
+        for (int i = 0; i < totalEvents; i++) {
+            writer.writeEvent(streamName + String.valueOf(i)).join();
+            log.debug("Writing event: {} to stream {}", streamName + String.valueOf(i), streamName);
+        }
+    }
+
+    private List<CompletableFuture<Integer>> readDummyEvents(ClientFactory client, String rGroup, int numReaders) {
+        List<EventStreamReader<String>> readers = new ArrayList<>();
+        for (int i = 0; i < numReaders; i++) {
+            readers.add(client.createReader(String.valueOf(i), rGroup, new JavaSerializer<>(), ReaderConfig.builder().build()));
+        }
+
+        return readers.stream().map(r -> CompletableFuture.supplyAsync(() -> readEvents(r))).collect(toList());
+    }
+
+    @SneakyThrows
+    private <T> int readEvents(EventStreamReader<T> reader) {
+        EventRead<T> event;
+        int validEvents = 0;
+        try {
+            do {
+                event = reader.readNextEvent(1000);
```

Comment:
Please add logging for read event as well. We have for writing above, but not for reading.

Reply:
Done.